Core/PaymentMethodHelper.php

```diff
@@ -129,6 +136,174 @@ public static function getBic()
+     *
+     * @param string $sDbDateOfBirth formated for db (format 'Y-m-d')
+     *
+     * @since 1.2.0
+     */
+    public static function setDbDateOfBirth($sDbDateOfBirth)
+    {
+        $oSession = Registry::getConfig()->getSession();
+        $aDynvalues = $oSession->getVariable('dynvalue');
+        $aDynvalues['dateOfBirth'] = '';
+
+        if ($sDbDateOfBirth !== self::DEFAULT_DATE_OF_BIRTH) {
+            $oDateOfBirth = DateTime::createFromFormat(self::DB_DATE_FORMAT, $sDbDateOfBirth);
+
+            if ($oDateOfBirth) {
+                $aDynvalues['dateOfBirth'] =
+                    $oDateOfBirth->format(Helper::translate('wd_birthdate_format_php_code'));
+            }
+        }
+
+        $oSession->setVariable('dynvalue', $aDynvalues);
+    }
+
+    /**
+     * Returns true if a valid date of birth is available
+     *
+     * @return bool
+     *
+     * @since 1.2.0
+     */
+    public static function isDateOfBirthSet()
+    {
+        return PaymentMethodHelper::getDbDateOfBirth() !== self::DEFAULT_DATE_OF_BIRTH;
+    }
+
+    /**
+     * Returns true if user is min 18 years old or date of birth is not known
+     *
+     * @return bool
+     *
+     * @since 1.2.0
+     */
+    public static function isUserEighteen()
+    {
+        $oSession = Registry::getConfig()->getSession();
+        $aDynvalues = $oSession->getVariable('dynvalue');
+
+        $oDateOfBirth =
+            DateTime::createFromFormat(Helper::translate('wd_birthdate_format_php_code'), $aDynvalues['dateOfBirth']);
+
+        if (!$oDateOfBirth) {
+            return true;
+        }
+
+        $oToday = new DateTime();
+        $oDateInterval = $oDateOfBirth->diff($oToday);
+
+        return $oDateInterval->invert === 0 && $oDateInterval->y >= 18;
+    }
+
+    /**
+     * Returns phone
+     *
+     * @return string
+     *
+     * @since 1.2.0
+     */
+    public static function getPhone()
+    {
+        $oSession = Registry::getConfig()->getSession();
+        $aDynvalues = $oSession->getVariable('dynvalue');
+        $sPhone = $aDynvalues['phone'];
+
+        return $sPhone ? $sPhone : '';
+    }
+
+    /**
+     * Sets the phone number
+     *
+     * @param string $sPhone
+     *
+     * @since 1.2.0
+     */
+    public static function setPhone($sPhone)
+    {
+        $oSession = Registry::getConfig()->getSession();
+        $aDynvalues = $oSession->getVariable('dynvalue');
+        $aDynvalues['phone'] = $sPhone;
+
+        $oSession->setVariable('dynvalue', $aDynvalues);
+    }
+
+    /**
+     * Returns true if a valid phone number is available or not needed
+     *
+     * @return bool
+     *
+     * @since 1.2.0
+     */
+    public static function isPhoneValid()
+    {
+        return !self::isPhoneNeeded() || PaymentMethodHelper::getPhone() !== '';
+    }
+
+    /**
+     * Returns true if a phone number is needed
+     *
+     * @return bool
+     *
+     * @since 1.2.0
+     */
+    public static function isPhoneNeeded()
```

Comment:
I think this is too specific for a helper.

Reply:
yes I will change it

Reply:
it is not needed for gi by wirecard, I removed it and will handle it in gi by payolution in the payment method itself